test/Hystrix.Dotnet.UnitTests/HystrixRollingPercentileTests.cs

```diff
@@ -92,16 +110,20 @@ public class GetMean
             [Fact]
             public void Returns_The_Mean_Of_All_Values_In_The_Snapshot_When_GetMetricsRollingPercentileEnabled_Is_True()
             {
+                var dateTimeProvider = new Mock<IDateTimeProvider>();
                 var timeInMilliseconds = 10000;
                 var numberOfBuckets = 10;
                 var bucketDataLength = 100;
                 var configurationServiceMock = new Mock<IHystrixConfigurationService>();
                 configurationServiceMock.Setup(x => x.GetMetricsRollingPercentileEnabled()).Returns(true);
-                var rollingPercentile = new HystrixRollingPercentile(timeInMilliseconds, numberOfBuckets, bucketDataLength, configurationServiceMock.Object);
+
+                dateTimeProvider.SetupGet(d => d.CurrentTimeInMilliseconds).Returns(0);
+
+                var rollingPercentile = new HystrixRollingPercentile(dateTimeProvider.Object, timeInMilliseconds, numberOfBuckets, bucketDataLength, configurationServiceMock.Object);
                 rollingPercentile.AddValue(243);
                 rollingPercentile.AddValue(157);
 
-                Thread.Sleep(1500);
```

Comment:
This was pretty ugly before, if I understood it correctly, we were doing a `Thread.Sleep()` in the unit test, and actually depending on the proper amount of time being passed. I changed it to use mocked times.